src/Service/Indexing/IndexingEvents.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+namespace App\Service\Indexing;
+
+use App\Exception\IndexingException;
+use App\Model\Indexing\IndexFieldTypes;
+use App\Model\Indexing\IndexNames;
+use App\Service\ImageHandlerInterface;
+use Elastic\Elasticsearch\Client;
+use Elastic\Elasticsearch\Exception\ClientResponseException;
+use Elastic\Elasticsearch\Exception\MissingParameterException;
+use Elastic\Elasticsearch\Exception\ServerResponseException;
+use Elastic\Elasticsearch\Response\Elasticsearch;
+use Symfony\Component\DependencyInjection\Attribute\AsTaggedItem;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Serializer\Context\Normalizer\DateTimeNormalizerContextBuilder;
+use Symfony\Component\Serializer\Context\Normalizer\ObjectNormalizerContextBuilder;
+use Symfony\Component\Serializer\SerializerInterface;
+
+#[AsTaggedItem(index: IndexNames::Events->value, priority: 10)]
+final class IndexingEvents extends AbstractIndexingElastic
+{
+    public function __construct(
+        private readonly SerializerInterface $serializer,
+        private readonly ImageHandlerInterface $imageHandler,
+        private readonly string $indexAliasName,
+        private readonly Client $client,
+    ) {
+        parent::__construct($this->indexAliasName, $this->client);
+    }
+
+    public function serialize(IndexItemInterface $item): array
+    {
+        $contextBuilder = (new ObjectNormalizerContextBuilder())
+            ->withGroups([IndexNames::Events->value]);
+        $contextBuilder = (new DateTimeNormalizerContextBuilder())
+            ->withContext($contextBuilder)
+            ->withFormat(IndexFieldTypes::DATEFORMAT);
+        $data = $this->serializer->normalize($item, null, $contextBuilder->toArray());
+
+        // @todo: Figure out how to do these changes with the serializer. This is just....
+        // Flatten tags.
+        $data['tags'] = array_map(fn ($tag) => $tag['name'], $data['tags']);
+
+        // Flatten location address and convert lang/long to coordinate point.
+        $data['location'] += $data['location']['address'];
+        unset($data['location']['address']);
+        $data['location']['coordinates'] = [];
+        $data['location']['coordinates'] = [$data['location']['latitude'], $data['location']['longitude']];
+        unset($data['location']['latitude']);
+        unset($data['location']['longitude']);
+
+        // Fix image urls (with a full path and derived sizes).
+        $imageUrl = $data['imageUrl']['original'];
+        $data['imageUrl'] = $this->imageHandler->getDerived($imageUrl);
```

Comment:
`src/Service/ImageHandlerInterface::getDerived` returns an array:

```suggestion
        $data['imageUrls'] = $this->imageHandler->getDerived($imageUrl);
```

Reply:
Fixed